tools/generator/page-builder-admin.js

```diff
@@ -0,0 +1,635 @@
+  }
+
+  async loadAllDataSources() {
+    await Promise.all(this.dataSources.map((s) => this.loadDataSource(s.id)));
+  }
+
+  buildMergedItems(source) {
+    const items = [];
+    Object.entries(source.data).forEach(([sheetName, sheetData]) => {
+      Object.entries(sheetData).forEach(([key, item]) => {
+        items.push({
+          currentValue: item.stored?.value || item.live?.value || '',
+          currentLabel: item.stored?.label || item.live?.label || '',
+          state: getItemState(item),
+          itemKey: `${sheetName}:${key}`,
+          sheetName,
+          live: item.live,
+          stored: item.stored,
+          metadata: item.metadata,
+          blocked: item.metadata?.blocked || false,
+        });
+      });
+    });
+    return items;
+  }
+
+  getCurrentSheetItems() {
+    return this.activeSheet ? this.mergedItems.filter((m) => m.sheetName === this.activeSheet) : this.mergedItems;
+  }
+
+  isSheetBlocked(sheetName) {
+    return this.blockedSheets.has(sheetName);
+  }
+
+  handleEdit(sourceId) {
+    const source = this.dataSources.find((s) => s.id === sourceId);
+    if (!source) return;
+
+    this.editingSource = sourceId;
+    this.mergedItems = this.buildMergedItems(source);
+    this.blockedSheets = getBlockedSheets(source.metadata);
+    this.pendingChanges = new Set();
+
+    const sheets = Object.keys(source.data);
+    this.activeSheet = sheets.length > 0 ? sheets[0] : null;
+    this.requestUpdate();
+  }
+
+  handleCancelEdit() {
+    const sourceId = this.editingSource;
+    this.editingSource = null;
+    this.mergedItems = [];
+    this.activeSheet = null;
+    this.blockedSheets = new Set();
+    this.pendingChanges = new Set();
+    this.requestUpdate();
+    if (sourceId) this.loadDataSource(sourceId);
+  }
+
+  handleSheetChange(sheetName) {
+    this.activeSheet = sheetName;
+    this.requestUpdate();
+  }
+
+  handleSyncItem(itemKey) {
+    const item = this.mergedItems.find((m) => m.itemKey === itemKey);
+    if (!item?.live) return;
+
+    item.currentValue = item.live.value;
+    item.currentLabel = item.live.label;
+    item.state = ITEM_STATES.SYNCED;
+    this.pendingChanges.add(itemKey);
+    this.requestUpdate();
+  }
+
+  handleToggleBlock(itemKey) {
+    const item = this.mergedItems.find((m) => m.itemKey === itemKey);
+    if (!item) return;
+
+    item.blocked = !item.blocked;
+    if (!item.metadata) item.metadata = {};
+    item.metadata.blocked = item.blocked;
+    item.state = getItemState({ live: item.live, stored: item.stored, metadata: item.metadata });
+    this.pendingChanges.add(itemKey);
+    this.requestUpdate();
+  }
+
+  handleRemoveItem(itemKey) {
+    this.mergedItems = this.mergedItems.filter((m) => m.itemKey !== itemKey);
+    this.pendingChanges.add(itemKey);
+    this.requestUpdate();
+  }
+
+  syncItems(items) {
+    const syncable = items.filter((i) => i.live && !i.blocked && [ITEM_STATES.NEW, ITEM_STATES.DRIFTED].includes(i.state));
```

Comment:
nit: change the var name to `syncableItems` so we are aware the return state is an iterable.